src/output.cpp

```diff
@@ -0,0 +1,736 @@
+               MPI_SUM, 0, mpi::intracomm);
+  #endif
+
+  if (mpi::master) {
+    header("cell overlap check summary", 1);
+    std::cout << " Cell ID      No. Overlap Checks\n";
+
+    std::vector<int32_t> sparse_cell_ids;
+    for (int i = 0; i < model::cells.size(); i++) {
+      std::cout << " " << std::setw(8) << model::cells[i]->id_ << std::setw(17)
+                << model::overlap_check_count[i] << "\n";
+      if (model::overlap_check_count[i] < 10) {
+        sparse_cell_ids.push_back(model::cells[i]->id_);
+      }
+    }
+
+    std::cout << "\n There were " << sparse_cell_ids.size()
+              << " cells with less than 10 overlap checks\n";
+    for (auto id : sparse_cell_ids) {
+      std::cout << " " << id;
+    }
+    std::cout << "\n";
+  }
+}
+
+//==============================================================================
+
+void print_usage()
+{
+  if (mpi::master) {
+    std::cout <<
+      "Usage: openmc [options] [directory]\n\n"
+      "Options:\n"
+      "  -c, --volume           Run in stochastic volume calculation mode\n"
+      "  -g, --geometry-debug   Run with geometry debugging on\n"
+      "  -n, --particles        Number of particles per generation\n"
+      "  -p, --plot             Run in plotting mode\n"
+      "  -r, --restart          Restart a previous run from a state point\n"
+      "                         or a particle restart file\n"
+      "  -s, --threads          Number of OpenMP threads\n"
+      "  -t, --track            Write tracks for all particles\n"
+      "  -v, --version          Show version information\n"
+      "  -h, --help             Show this message\n";
+  }
+}
+
+//==============================================================================
+
+void print_version()
+{
+  if (mpi::master) {
+    std::cout << "OpenMC version " << VERSION_MAJOR << '.' << VERSION_MINOR
+      << '.' << VERSION_RELEASE << '\n';
+#ifdef GIT_SHA1
+    std::cout << "Git SHA1: " << GIT_SHA1 << '\n';
+#endif
+    std::cout << "Copyright (c) 2011-2019 Massachusetts Institute of "
+      "Technology and OpenMC contributors\nMIT/X license at "
+      "<http://openmc.readthedocs.io/en/latest/license.html>\n";
+  }
+}
+
+//==============================================================================
+
+void print_columns()
+{
+  if (settings::entropy_on) {
+    std::cout <<
+      "  Bat./Gen.      k       Entropy         Average k \n"
+      "  =========   ========   ========   ====================\n";
+  } else {
+    std::cout <<
+      "  Bat./Gen.      k            Average k\n"
+      "  =========   ========   ====================\n";
+  }
+}
+
+//==============================================================================
+
+void print_generation()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = overall_generation() - 1;
+  int n = simulation::current_batch > settings::n_inactive ?
```

Comment:
Nit / Optional: Recommend changing `n` to active_generation_count or `n_active_generations`.